packages/next-international/src/app/middleware/index.ts

```diff
@@ -13,25 +13,26 @@ export function createI18nMiddleware<const Locales extends readonly string[]>(co
     const locale = localeFromRequest(config.locales, request, config.resolveLocaleFromRequest) ?? config.defaultLocale;
 
     if (noLocalePrefix(config.locales, requestUrl.pathname)) {
-      const mappedUrl = requestUrl.clone();
-      mappedUrl.pathname = `/${locale}${mappedUrl.pathname}`;
+      requestUrl.pathname = `/${locale}${requestUrl.pathname}`;
 
       const strategy = config.urlMappingStrategy ?? DEFAULT_STRATEGY;
-
       if (strategy === 'rewrite' || (strategy === 'rewriteDefault' && locale === config.defaultLocale)) {
-        const response = NextResponse.rewrite(mappedUrl);
+        const response = NextResponse.rewrite(requestUrl);
+        cloneCookies(request, response);
         return addLocaleToResponse(response, locale);
       } else {
         if (!['redirect', 'rewriteDefault'].includes(strategy)) {
           warn(`Invalid urlMappingStrategy: ${strategy}. Defaulting to redirect.`);
         }
 
-        const response = NextResponse.redirect(mappedUrl);
+        const response = NextResponse.redirect(requestUrl);
+        cloneCookies(request, response);
```

Comment:
We can probably move the `cloneCookies` call inside `addLocaleToResponse` to avoid duplicating it in many places.

Reply:
I've moved the `cloneCookies` code to `addLocaleToResponse`.
Thanks